controllers/provisioners/eksfargate/eksfargate.go

```diff
@@ -0,0 +1,290 @@
+		if err != nil {
+			ctx.Log.Info("Failed to get attach policy to role", "error", err)
+			return err
+		}
+		ctx.Log.Info("Attached default policy to role")
+
+	} else {
+		arn = spec.GetPodExecutionRoleArn()
+	}
+
+	ctx.Log.Info("Creating a profile.", "arn", arn)
+	tryAgain, err := ctx.AwsWorker.CreateProfile(arn)
+	if tryAgain {
+		ctx.Log.Info("Resource inuse on Create.")
+		return nil
+	}
+	if err != nil {
+		ctx.Log.Info("Creation of the fargate profile failed", "cluster",
+			spec.GetClusterName(),
+			"profile",
+			spec.GetProfileName(),
+			"error", err)
+	} else {
+		instanceGroup.SetState(v1alpha1.ReconcileModifying)
+	}
+
+	return err
+}
+func (ctx *InstanceGroupContext) CloudDiscovery() error {
+	profile, err := ctx.AwsWorker.DescribeProfile()
+	if err != nil {
+		profile = &eks.FargateProfile{
+			Status: nil,
+		}
+	}
+	if profile.Status == nil {
+		ctx.DiscoveredState.ProfileStatus = aws.FargateProfileStatusMissing
+	} else {
+		ctx.DiscoveredState.ProfileStatus = *profile.Status
+	}
+	return nil
+}
+func (ctx *InstanceGroupContext) Delete() error {
+	instanceGroup := ctx.GetInstanceGroup()
+	spec := instanceGroup.GetEKSFargateSpec()
+
+	worker := ctx.AwsWorker
+	if spec.GetPodExecutionRoleArn() == "" {
+		found, err := worker.DetachDefaultPolicyFromDefaultRole()
+		if found || err != nil {
+			ctx.Log.Info("Detaching the default policy", "error", err)
+			return err
+		}
+
+		found, err = ctx.AwsWorker.DeleteDefaultRole()
+		if found || err != nil {
+			ctx.Log.Info("Deleting the default role", "error", err)
+			return err
+		}
+	}
+	ctx.Log.Info("Deleting the profile")
+	tryAgain, err := worker.DeleteProfile()
+	if tryAgain {
+		ctx.Log.Info("Resource inuse on Delete.")
+		return nil
+	}
+	if err != nil {
+		ctx.Log.Info("Deletion of the fargate profile.", "cluster",
+			spec.GetClusterName(),
+			"profile",
+			spec.GetProfileName(),
+			"error", err)
+		return err
+	}
+	instanceGroup.SetState(v1alpha1.ReconcileDeleting)
+
+	return err
+}
+
+func (ctx *InstanceGroupContext) Update() error {
+	// No update is required
+	ctx.Log.Info("Running update")
+	instanceGroup := ctx.GetInstanceGroup()
+	annos := instanceGroup.GetObjectMeta().GetAnnotations()
+	// If there is a last-applied-configuration then assume
+	// this is an update and throw an exception
+	if _, ok := annos[LastAppliedConfigurationKey]; ok {
+		return errors.New("update not supported")
+	}
+	instanceGroup.SetState(v1alpha1.ReconcileModified)
+	return nil
+}
+func (ctx *InstanceGroupContext) UpgradeNodes() error {
+	return errors.New("upgrade not supported")
+}
+func (ctx *InstanceGroupContext) BootstrapNodes() error {
+	return nil
+}
+func (ctx *InstanceGroupContext) IsReady() bool {
+	instanceGroup := ctx.GetInstanceGroup()
+	if instanceGroup.GetState() == v1alpha1.ReconcileModified || instanceGroup.GetState() == v1alpha1.ReconcileDeleted {
+		return true
+	}
+	return false
+}
+func (ctx *InstanceGroupContext) IsUpgradeNeeded() bool {
```

Comment:
This is not a required method in the interface, you can probably remove it

Reply:
will do.

Reply:
I love removing code.  Plus i get rid of a test.